src/main/java/org/assertj/core/api/AbstractSoftAssertions.java

```diff
@@ -31,6 +31,21 @@ public AbstractSoftAssertions() {
     return proxies.createSoftAssertionProxy(assertClass, actualClass, actual);
   }
 
+  /**
+   * Catch and collect assertion errors of any assertion call.
+   *
+   * @param assertion an assertion call.
+   */
+  public void check(ThrowingRunnable assertion) {
+    try {
+      assertion.run();
+    } catch (AssertionError error) {
+      proxies.collectError(error);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
it would be nice to catch runtime and rethrow them as is instead wrapping in them in another runtime.